snap_plugin/v1/metric.py

```diff
@@ -0,0 +1,259 @@
+# -*- coding: utf-8 -*-
+# http://www.apache.org/licenses/LICENSE-2.0.txt
+#
+# Copyright 2016 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import time
+from builtins import int
+
+from past.builtins import basestring
+
+from .config_map import ConfigMap
+from .namespace import Namespace
+from .plugin_pb2 import Metric as PbMetric
+from .timestamp import Timestamp
+
+
+class Metric(object):
+    """Metric
+
+    The metric object is the core entity of snap representing metrics.
+
+    Args:
+        namespace (:obj:`list` of
+            :py:class:`~snap_plugin.v1.namespace_element.NamespaceElement`):
+            namespace elements
+        version (:obj:`int`): metric version
+        tags (:obj:`dict`): metric tags (key/value pairs)
+        config (:py:class:`~snap_plugin.v1.config_map.ConfigMap`): config for
+            metric (key/value pairs)
+        timestamp (:obj:`float`): metric timestamp (see time.time())
+        unit (:obj:`str`): metric unit
+        description (:obj:`str`): metric description
+
+    """
+    def __init__(self, namespace=[], version=0, tags={}, config={},
```

Comment:
Similarly here, I think giving plugin author ability to build metric with Namespace instance or list of strings would be much easier. Providing list of NamespaceElements in Metric constructor seems little bit too verbose and complicated from the perspective of plugin author. What is your opinion?